packages/cli/src/commands/devicepreferences/create.ts

```diff
@@ -0,0 +1,167 @@
+		const title = await askForRequiredString('Preference title:')
+		const description = await askForString('Preference description:')
+
+		const required = (await inquirer.prompt({
+			type: 'confirm',
+			name: 'value',
+			message: 'Is the preference required?',
+			default: false,
+		})).value as boolean
+
+		const preferenceType = (await inquirer.prompt({
+			type: 'list',
+			name: 'preferenceType',
+			message: 'Choose a type for your preference.',
+			choices: ['integer', 'number', 'boolean', 'string', 'enumeration'],
+		})).preferenceType as PreferenceType
+
+		const base = {
+			name, title, description, required,
+		}
+
+		if (preferenceType === 'integer') {
+			const minimum = await askForInteger('Optional minimum value.')
+			const maximum = await askForInteger('Optional maximum value.', minimum)
+			const defaultValue = await askForInteger('Optional default value.', minimum, maximum)
+			return {
+				...base, preferenceType, definition: {
+					minimum: minimum ?? undefined,
+					maximum: maximum ?? undefined,
+					default: defaultValue ?? undefined,
+				},
+			}
+		}
+
+		if (preferenceType === 'number') {
+			const minimum = await askForNumber('Optional minimum value.')
+			const maximum = await askForNumber('Optional maximum value.', minimum)
+			const defaultValue = await askForNumber('Optional default value.', minimum, maximum)
+			return {
+				...base, preferenceType, definition: {
+					minimum: minimum ?? undefined,
+					maximum: maximum ?? undefined,
+					default: defaultValue ?? undefined,
+				},
+			}
+		}
+
+		if (preferenceType === 'boolean') {
+			const defaultValue = (await inquirer.prompt({
+				type: 'list',
+				name: 'defaultValue',
+				message: 'Choose a default value.',
+				choices: [{ name: 'none', value: undefined },
+					{ name: 'true', value: true },
+					{ name: 'false', value: false }],
+			})).defaultValue as boolean | undefined
+			return {
+				...base, preferenceType, definition: {
+					default: defaultValue ?? undefined,
+				},
+			}
+		}
+
+		if (preferenceType === 'string') {
+			const minLength = await askForInteger('Optional minimum length.')
+			const maxLength = await askForInteger('Optional maximum length.', minLength)
+			const stringType = (await inquirer.prompt({
+				type: 'list',
+				name: 'stringType',
+				message: 'Choose a type of string.',
+				choices: ['text', 'password', 'paragraph'],
+				default: 'text',
+			})).stringType as 'text' | 'password' | 'paragraph'
+			const defaultValue = await askForString('Optional default value.',
+				input => {
+					if (minLength !== undefined && input.length < minLength) {
+						return `default must be at least minLength (${minLength}) characters`
+					}
+					if (maxLength !== undefined && input.length > maxLength) {
+						return `default must be no more than maxLength (${maxLength}) characters`
```

Comment:
We should make these (and the similar messaging in `user-query.ts`) consistent and both some form of `no more/no less than` or `at most/at least`. I don't have strong feelings either way.

Reply:
changed